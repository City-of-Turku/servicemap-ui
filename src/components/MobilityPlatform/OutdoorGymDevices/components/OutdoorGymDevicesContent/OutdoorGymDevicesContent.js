import PropTypes from 'prop-types';
import React from 'react';
import { StyledContainer, StyledHeaderContainer, StyledTextContainer } from '../../../styled/styled';
import TextComponent from '../../../TextComponent';

const OutdoorGymDevicesContent = ({ item }) => {
  const deviceName = {
    fi: item.name_fi,
    en: item.name_en,
    sv: item.name_sv,
  };

  const deviceAddress = {
    fi: item.address_fi,
    en: item.address_en,
    sv: item.address_sv,
  };

  const deviceDescription = {
    fi: item.description_fi,
    en: item.description_en,
    sv: item.description_sv,
  };

  return (
    <StyledContainer>
      <StyledHeaderContainer>
        <TextComponent textObj={deviceName} isTitle />
      </StyledHeaderContainer>
      <StyledTextContainer>
        {item.address_fi !== '' ? <TextComponent messageId="mobilityPlatform.content.address" textObj={deviceAddress} /> : null}
        <TextComponent textObj={deviceDescription} />
      </StyledTextContainer>
    </StyledContainer>
  );
};

OutdoorGymDevicesContent.propTypes = {
  item: PropTypes.shape({
    address_fi: PropTypes.string,
    address_en: PropTypes.string,
    address_sv: PropTypes.string,
    name_fi: PropTypes.string,
    name_en: PropTypes.string,
    name_sv: PropTypes.string,
    description_fi: PropTypes.string,
    description_en: PropTypes.string,
    description_sv: PropTypes.string,
  }),
};

OutdoorGymDevicesContent.defaultProps = {
  item: {},
};

export default OutdoorGymDevicesContent;
