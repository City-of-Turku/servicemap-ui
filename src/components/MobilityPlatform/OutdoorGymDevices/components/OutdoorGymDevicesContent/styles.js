export default theme => ({
  container: {
    margin: theme.spacing(1),
  },
  headerContainer: {
    width: '85%',
    borderBottom: '1px solid #000',
    paddingBottom: theme.spacing(0.5),
  },
  textContainer: {
    marginTop: theme.spacing(0.5),
  },
  contentInner: {
    marginLeft: theme.spacing(1),
    marginBottom: theme.spacing(1),
  },
  margin: {
    margin: theme.spacing(0.4),
  },
  marginTop: {
    marginTop: theme.spacing(0.4),
  },
});
