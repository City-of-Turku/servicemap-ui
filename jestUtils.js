import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import finnish from './src/i18n/fi';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnish,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });