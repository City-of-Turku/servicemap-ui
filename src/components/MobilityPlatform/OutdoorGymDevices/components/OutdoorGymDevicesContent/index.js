import { withStyles } from '@mui/styles';
import OutdoorGymDevicesContent from './OutdoorGymDevicesContent';
import styles from './styles';

export default withStyles(styles)(OutdoorGymDevicesContent);
