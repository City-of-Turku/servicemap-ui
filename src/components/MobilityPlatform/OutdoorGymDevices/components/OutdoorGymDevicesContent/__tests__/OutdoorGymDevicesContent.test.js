// Link.react.test.js
import React from 'react';
import OutdoorGymDevicesContent from '../index';
import { initialState } from '../../../../../../redux/reducers/user';
import { getRenderWithProviders } from '../../../../../../../jestUtils';

const mockProps = {
  item: {
    name_fi: 'Testi',
    name_en: 'Test',
    name_sv: 'Test',
    address_fi: 'Testikatu',
    address_en: 'Test street',
    address_sv: 'Test gata',
    description_fi: 'Testikuvaus',
    description_en: 'Test description',
    description_sv: 'Test skrivning',
  },
};

const renderWithProviders = getRenderWithProviders({
  user: initialState,
});

describe('<OutdoorGymDevicesContent />', () => {
  it('should match snapshot', () => {
    const { container } = renderWithProviders(<OutdoorGymDevicesContent {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<OutdoorGymDevicesContent {...mockProps} />);
    const p = container.querySelectorAll('p');
    expect(p[0].textContent).toContain(mockProps.item.name_fi);
    expect(p[1].textContent).toContain(`Osoite: ${mockProps.item.address_fi}`);
    expect(p[2].textContent).toContain(mockProps.item.description_fi);
  });
});
