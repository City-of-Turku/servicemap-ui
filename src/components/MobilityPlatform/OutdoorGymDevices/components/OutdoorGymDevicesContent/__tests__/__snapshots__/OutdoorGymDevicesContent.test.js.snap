// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OutdoorGymDevicesContent /> should match snapshot 1`] = `
<div>
  <div
    class="OutdoorGymDevicesContent-container-1"
  >
    <div
      class="OutdoorGymDevicesContent-headerContainer-2"
    >
      <div
        class="injectIntl(TextComponent)-margin-7"
      >
        <p
          class="MuiTypography-root MuiTypography-subtitle1"
        >
          Testi
        </p>
      </div>
    </div>
    <div
      class="OutdoorGymDevicesContent-textContainer-3"
    >
      <div
        class="injectIntl(TextComponent)-margin-7"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Osoite: Testikatu
        </p>
      </div>
      <div
        class="injectIntl(TextComponent)-margin-7"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Testikuvaus
        </p>
      </div>
    </div>
  </div>
</div>
`;
